packages/next-server/server/next-server.ts

```diff
@@ -226,6 +234,32 @@ export default class Server {
     return routes
   }
 
+  /**
+   * Resolves `API` request, in development builds on demand
+   * @param req http request
+   * @param res http response
+   * @param pathname path of request
+   */
+  private async handleApiRequest(req: IncomingMessage, res: ServerResponse, pathname: string) {
+    const resolverFunction = await this.resolveApiRequest(pathname)
+    if (resolverFunction === null) {
+      res.statusCode = 501
```

Comment:
This should be 404. When API route doesn't exist it should set 404 and text should be `Not Found` I guess.